qemu/deps/playbook.yml

```diff
@@ -40,18 +39,34 @@
         /usr/bin/update-ca-trust extract
       when: dnld_cert_output.changed
 
-# The following two tasks need to be modified, check if kar already deployed, if true, skip.
-    - name: clone kar
+    - name: Check kar folder exists
+      stat:
+         path: '{{kar_local}}'
+      register: kar_folder
+
+    - name: Clone kar
       git:
         repo: '{{kar_repo}}'
         dest: '{{kar_local}}'
         force: yes
+      when: not kar_folder.stat.exists
 
     - name: Run bootstrap script
       command: ./Bootstrap.sh {{bs_options}}
       args:
         chdir: '{{kar_local}}'
+      when: not kar_folder.stat.exists
       register: bootstrap_result
+      ignore_errors: True
+
+    - name: Clean env when bootstrap failed
+      shell: |
+        rm -rf '{{kar_local}}'
+        pip uninstall -y `pip list | grep avocado`
+        pip uninstall -y `pip list | grep aexpect`
```

Comment:
If these package still have not be installed before get `bootstrap_result` failed, will get error: `ERROR: You must give at least one requirement to uninstall`, suggest use the 'pip' module for ansible:
```
      - name: clean env ****
        pip:
          name: aexpect, avocado-framework, avocado-framework-plugin-vt
          state: absent
```

Reply:
If use pip, we need to remove kar directory in another task. This is not expected.

Reply:
> If use pip, we need to remove kar directory in another task. This is not expected.

Split to two task is ok, it is not must to put them in one task i think.  Just clean the env is ok.